src/Symfony/Bridge/Monolog/Tests/Processor/WebProcessorTest.php

```diff
@@ -20,6 +20,38 @@ class WebProcessorTest extends \PHPUnit_Framework_TestCase
 {
     public function testUsesRequestServerData()
     {
+        list($event, $server) = $this->createRequestEvent();
+
+        $processor = new WebProcessor();
+        $processor->onKernelRequest($event);
+        $record = $processor($this->getRecord());
+
+        $this->assertCount(5, $record['extra']);
+        $this->assertEquals($server['REQUEST_URI'], $record['extra']['url']);
+        $this->assertEquals($server['REMOTE_ADDR'], $record['extra']['ip']);
+        $this->assertEquals($server['REQUEST_METHOD'], $record['extra']['http_method']);
+        $this->assertEquals($server['SERVER_NAME'], $record['extra']['server']);
+        $this->assertEquals($server['HTTP_REFERER'], $record['extra']['referrer']);
+    }
+
+    public function testCanBeConstructedWithExtraFields()
+    {
+        list($event, $server) = $this->createRequestEvent();
+
+        $processor = new WebProcessor(array('url', 'referrer'));
+        $processor->onKernelRequest($event);
+        $record = $processor($this->getRecord());
+
+        $this->assertCount(2, $record['extra']);
+        $this->assertEquals($server['REQUEST_URI'], $record['extra']['url']);
+        $this->assertEquals($server['HTTP_REFERER'], $record['extra']['referrer']);
+    }
+
+    /**
+     * @return array
+     */
+    protected function createRequestEvent()
```

Comment:
I'm neutral in this case. `createRequestEvent()` is `protected` because `getRecored()` is protected. I could change my method to `private` or both, but changing `protected` to `private` could be considered BC, so I'm not going to decide which way should it go.